### PR TITLE
Validate constant expressions in globals initializers

### PR DESCRIPTION
**Pull request by gumb0**
First part of https://github.com/wasmx/fizzy/pull/372

### CODE REVIEW

File: test/unittests/parser_test.cpp [hunk 1]
```diff
@@ -594,10 +594,15 @@ TEST(parser, global_single_mutable_const_inited)
 
 TEST(parser, global_single_const_global_inited)
```
Comment (axic) [inline test/unittests/parser_test.cpp:595]: Test name doesn't reflect the test anymore.

Comment (gumb0) [reply test/unittests/parser_test.cpp]: Renamed

File: lib/fizzy/parser.cpp [hunk 2]
```diff
@@ -12,6 +12,22 @@
...
+{
+    if (const_expr.kind == ConstantExpression::Kind::Constant)
+        return;
+
+    const auto global_idx = const_expr.value.global_index;
+    if (global_idx >= module.get_global_count())
+        throw validation_error{"invalid global index in constant expression"};
+
+    if (module.is_global_mutable(global_idx))
+        throw validation_error("constant expression can use global.get only for const globals");
```
Comment (axic) [inline lib/fizzy/parser.cpp:27]: Why is this using `validation_error()` while the other `validation_error{}` ?

File: lib/fizzy/parser.cpp [hunk 3]
```diff
@@ -531,11 +547,26 @@ Module parse(bytes_view input)
...
+    for (const auto& global : module.globalsec)
+    {
+        validate_constant_expression(global.expression, module);
+
+        // Wasm spec section 3.3.7 constrains initialization by another global to const imports only
+        // https://webassembly.github.io/spec/core/valid/instructions.html#expressions
+        if (global.expression.kind == ConstantExpression::Kind::GlobalGet &&
+            global.expression.value.global_index >= module.imported_globals_mutability.size())
+        {
+            throw validation_error(
```
Comment (axic) [inline lib/fizzy/parser.cpp:560]: Also uses parenthesis. It seems we are inconsistently using parentheses and curly brackets.

File: lib/fizzy/parser.cpp (hunk 2 above)
Comment (gumb0) [reply lib/fizzy/parser.cpp]: Changed it to always curly braces in my changes.

**Comment by gumb0** [comment]
Rebased.

**Comment by axic** [comment]
Rebased. @gumb0 please check.

**Pull request merged**